napari/_vispy/overlays/interaction_box.py

```diff
@@ -107,12 +134,25 @@ def _compute_vertices_and_box(self):
 
             edge_color = self._highlight_color
             if self._interaction_box.show_vertices:
+                colors = np.array([(1.0, 1.0, 1.0) for point in box])
+                if self._interaction_box.selected_vertex is not None:
+                    colors[
+                        self._interaction_box.selected_vertex
+                    ] = self._highlight_color
                 if self._interaction_box.show_handle:
                     vertices = box[Box.WITH_HANDLE][:, ::-1]
+                    face_color = ColorArray(
+                        [color for color in colors[Box.WITH_HANDLE]]
```

Comment:
I think vispy's `ColorArray` can take a numpy array as input, so I think you can do the following instead.

```suggestion
                    face_color = ColorArray(colors[Box.WITH_HANDLE]
```